mono/metadata/icall.c

```diff
@@ -1238,6 +1242,15 @@ ves_icall_System_ValueType_Equals (MonoObject *this_obj, MonoObject *that, MonoA
 			if (*(double *) this_field != *(double *) that_field)
 				return FALSE;
 			break;
+		case MONO_TYPE_PTR:
+#ifdef NO_UNALIGNED_ACCESS
+			if (G_UNLIKELY ((intptr_t) this_field & 7 || (intptr_t) that_field & 7))
+				UNALIGNED_COMPARE (gpointer);
+			else
+#endif
```

Comment:
It would be better to duplicate the check inside both #ifdef cases.

Reply:
i.e.
else
  <check>
#else
  <check>
#endif

Reply:
Ie the if (*(gpointer *) this_field != *(gpointer *) that_field)
+				return FALSE;
lines